pkg/apply/taskrunner/task.go

```diff
@@ -244,6 +280,8 @@ func (w *WaitTask) StatusUpdate(taskContext *TaskContext, id object.ObjMetadata)
 	}
 
 	// if all conditions are met, complete the wait task
+	// TODO(mortent): Isn't it enough to just check the size of the
```

Comment:
You're probably right, assuming we received a StatusUpdate for every object, then we could just check if `len(pending) == 0`

Reply:
Can we assume we get a StatusUpdate message for every object? I assume the case where this might not hold would be resources that reconcile before the WaitTask is started?

Reply:
If it reconciled before the WaitTask started, we won't get a StatusUpdate for it unless the generation or status changes. But skipped and resolved objects aren't added to `pending` in `Start` so that shouldn't be a problem.

I think we can change it to  do `len(pending) == 0` instead. And that might make the `conditionMet` function obsolete, since we only need to check one resource at a time now.

Reply:
Updated the check to use `len(pending) == 0`. I think we can also do some cleanup of the `conditionMet` function and other functionality used by it. But I'll do that in a separate cleanup PR to keep this one small.